tonic-types/src/lib.rs

```diff
@@ -13,8 +23,391 @@
+    }
+
+    fn with_error_details_vec(
+        code: Code,
+        message: impl Into<String>,
+        details: Vec<ErrorDetail>,
+    ) -> Self {
+        let message: String = message.into();
+
+        let mut conv_details: Vec<Any> = Vec::with_capacity(details.len());
+
+        for error_detail in details.into_iter() {
+            match error_detail {
+                ErrorDetail::BadRequest(bad_req) => {
+                    conv_details.push(bad_req.into_any());
+                }
+            }
+        }
+
+        let status = pb::Status {
+            code: code as i32,
+            message: message.clone(),
+            details: conv_details,
+        };
+
+        tonic::Status::with_details(code, message, Bytes::from(status.encode_to_vec()))
+    }
+
+    fn check_error_details(&self) -> Result<ErrorDetails, DecodeError> {
+        let status = pb::Status::decode(self.details())?;
+
+        let mut details = ErrorDetails::new();
+
+        for any in status.details.into_iter() {
+            match any.type_url.as_str() {
+                BadRequest::TYPE_URL => {
+                    details.bad_request = Some(BadRequest::from_any(any)?);
+                }
+                _ => {}
+            }
+        }
+
+        Ok(details)
+    }
+
+    fn get_error_details(&self) -> ErrorDetails {
+        self.check_error_details().unwrap_or(ErrorDetails::new())
+    }
+
+    fn check_error_details_vec(&self) -> Result<Vec<ErrorDetail>, DecodeError> {
+        let status = pb::Status::decode(self.details())?;
+
+        let mut details: Vec<ErrorDetail> = Vec::with_capacity(status.details.len());
+
+        for any in status.details.into_iter() {
+            match any.type_url.as_str() {
+                BadRequest::TYPE_URL => {
+                    details.push(BadRequest::from_any(any)?.into());
+                }
+                _ => {}
+            }
+        }
+
+        Ok(details)
+    }
+
+    fn get_error_details_vec(&self) -> Vec<ErrorDetail> {
+        self.check_error_details_vec().unwrap_or(Vec::new())
+    }
+
+    fn get_details_bad_request(&self) -> Option<BadRequest> {
+        let status = pb::Status::decode(self.details()).ok()?;
+
+        for any in status.details.into_iter() {
+            match any.type_url.as_str() {
+                BadRequest::TYPE_URL => match BadRequest::from_any(any) {
+                    Ok(detail) => return Some(detail),
+                    Err(_) => {}
+                },
+                _ => {}
+            }
+        }
+
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use tonic::{Code, Status};
+
+    use super::{BadRequest, ErrorDetails, WithErrorDetails};
+
+    #[test]
+    fn gen_status_with_details() {
+        let mut err_details = ErrorDetails::new();
+
+        err_details.add_bad_request_violation("field", "description");
+
+        let fmt_details = format!("{:?}", err_details);
+
+        println!("{fmt_details}\n");
```

Comment:
do we need all these prints? If you want more printing you can use tracing where you then enable the filters at runtime via `RUST_LOG=... cargo test`. Though that may be overkill so maybe we can just remove them?

Reply:
You're right, no need for them there anymore. The adjustments commit removes them